content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+    static public void Main()
+    {
+  
+        // Creating HashSet
+     
+        HashSet<string> myhash1 = new HashSet<string>();
+  
+        // Add the elements in HashSet
+  
+        myhash1.Add("Ndung'u");
+        myhash1.Add("Elijah");
+
+        // Creating another HashSet
+     
+        HashSet<string> myhash2 = new HashSet<string>();
+        myhash2.Add("Section");
+        myhash2.Add("Git");
+   
+  
+        // Using ExceptWith method
+        myhash1.ExceptWith(myhash2);
+        foreach(var ele in myhash1)
+        {
+            Console.WriteLine(ele);
+        }
+    }
+}
+```
+
+We note that many operations available for the list are available for hash sets except that:
+
+* We CANNOT add an object at a specific position since the elements are not in any order.
+* We cannot replace an item for the same reason, although we can add and delete another.
+* Retrieving all the items is possible but the sequence is indeterminate.
+* It's meaningless to find what position an element is or rather to sort.
+
+### 3. Dictionaries
+Dictionaries are slightly different from lists and sets because instead of individual objects, they store pairs; a key and a value.
+The key contains the address to the pair. The value is the piece of information associated with the key.
+Dictionaries have a very wide market application for instance an address book in which keys are people's names, and the values are address, phone, email, etc.
+Duplicate keys are not permitted though values are. Also, a dictionary doesn't store items in order, but it can be sorted in order of keys to store a sorted dictionary. A simple program to illustrate:
+
+```
+using System;
+using System.Collections.Generic;  
+  
+class MyDictinary {
+    static public void Main () {
+          
+      
+        //  Creating a dictionary using Dictionary <TKey,TValue> class
+        Dictionary<int, string> My_dict1 = new Dictionary<int, string>(); 
+            
+          // Adding <key/value> pair in the Dictionary using Add() method
+          My_dict1.Add(1123, "Welcome");
+          My_dict1.Add(1124, "Hello");
+          My_dict1.Add(1125, "World");
+            
+          foreach(KeyValuePair<int, string> ele1 in My_dict1)
+          {
+              Console.WriteLine("{0} and {1}", ele1.Key, ele1.Value);
+          }
+          Console.WriteLine();
+            
+          // Creating another dictionary
+ 
+      Dictionary<string, string> My_dict2 = new Dictionary<string, string>(){
+                                  {"a.1", "CSharp"},
+                                  {"a.2", "Java"},
+                                {"a.3", "Python"} }; 
+           
+          foreach(KeyValuePair<string, string> ele2 in My_dict2)
+          {
+              Console.WriteLine("{0} and {1}", ele2.Key, ele2.Value);
+          }
+    }
+}
+```
+
+## Conclusion
+In summary, I have explained what are generics in c#, its advantages, and a complete programming example   
```

Comment:
> In summary, I have explained what are generics in c#, its advantages, and a complete programming example   
> I also explained the comparison between generic and non - generic programming where both can add, delete, populate and sort Generic List. Lists, dictionaries, and hash sets are automatically resized every time an item is added or deleted.

The first sentence is missing a period in the end. Also in the second sentence, change: *non - generic* to *non-generic*; there shouldn’t be spaces before and after the *-*. Also, there are a couple of grammatical errors that need to be fixed. Also, personally, I would prefer to say: *we went through* or *we learned* instead of *I explained*. The tone just sounds more like you are trying to help and educate someone and sounds friendlier in a way. To fix all these, I would say something like: